tensorflow/contrib/layers/python/layers/layers.py

```diff
@@ -1481,6 +1483,57 @@ def max_pool2d(inputs,
 
 
 @add_arg_scope
+def maxout(inputs,
+           num_units,
+           axis=None,
+           outputs_collections=None,
+           scope=None):
+  """Adds a maxout op from https://arxiv.org/abs/1302.4389
+
+    "Maxout Networks"
+
+    Ian J. Goodfellow, David Warde-Farley, Mehdi Mirza, Aaron Courville, Yoshua
+    Bengio
+
+
+  Max pooling is performed in given filter/channel dimension. This can also be
+  used after fully-connected layers to reduce number of features.
+
+
+  Args:
+    inputs: A Tensor on which maxout will be performed
+    num_units: Specifies how many features will remain after max pooling at the
+      channel dimension. This must be multiple of number of channels.
+    axis: The dimension where max pooling will be performed. Default is the
+      last dimension.
+    outputs_collections: The collections to which the outputs are added.
+    scope: Optional scope for name_scope.
+
+  Returns:
+    A `Tensor` representing the results of the pooling operation.
```

Comment:
please describe shape of output tensor.

also may be in description add a formula e.g. out[..., i, ..., j] = max(in[...]) (similar how https://www.tensorflow.org/versions/r0.12/api_docs/python/nn.html#conv2d)